aleph/logic/xref.py

```diff
@@ -32,6 +37,47 @@
 
 log = logging.getLogger(__name__)
 ORIGIN = "xref"
+MODEL = None
+FTM_VERSION_STR = f"ftm-{followthemoney.__version__}"
+
+
+@dataclass
+class Match:
+    score: float
+    method: str
+    entity: EntityProxy
+    collection_id: str
+    match: EntityProxy
+    entityset_ids: typing.Sequence[str]
+    doubt: typing.Optional[float] = None
+
+
+def _bulk_compare_ftm(proxies):
+    for left, right in proxies:
+        score = compare.compare(model, left, right)
+        yield score, None, FTM_VERSION_STR
+
+
+def _bulk_compare_ftmc_model(proxies):
+    for score, confidence in zip(*MODEL.predict_proba_std(proxies)):
+        yield score, confidence, MODEL.version
+
+
+def _bulk_compare(proxies):
+    if not proxies:
+        return
+    if settings.XREF_MODEL is None:
+        yield from _bulk_compare_ftm(proxies)
```

Comment:
Is it ok to add an explicit `return` after the `yield from` statements? I know the execution stops because the number of items we yield is controlled by the number of candidates we process in `_query_item`. But it was not very obvious to me when looking just at this function.

Reply:
Good catch! It's definitely better to add the return after these `yield from`'s to make the flow more explicit in case things change in the future.